wivrn: init at 0.15

PassiveLemon opened the pull request:
## Description of changes
Adds WiVRn
https://github.com/NixOS/nixpkgs/pull/278126#issuecomment-1975239007

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python311Packages.tf-keras: init at 2.16.0

PassiveLemon committed:
wivrn: init at 0.15

ckiee commented:
Continuation of https://github.com/NixOS/nixpkgs/pull/293058

ckiee reviewed (changes requested):
Looks like all the previous reviews have been more-or-less addressed so it's just up to someone testing it with a headset. 

Putting in a _request changes_ to prevent accidental merges until it's tested. I'll try to do it whenever I get my headset out.

MayNiklas committed:
papermc: fix update script

PassiveLemon commented:
As far as I can tell, WiVRn works fine, it's Steam and Proton that are causing problems: https://github.com/ValveSoftware/steam-runtime/issues/575

PassiveLemon commented:
How would I use a package that isn't merged into nixpkgs yet but is in a different PR? I would like to install `monado-vulkan-layers` from https://github.com/NixOS/nixpkgs/pull/301044#issuecomment-2033106270 in the module

Scrumplex commented:
> How would I use a package that isn't merged into nixpkgs yet but is in a different PR? I would like to install `monado-vulkan-layers` from [#301044 (comment)](https://github.com/NixOS/nixpkgs/pull/301044#issuecomment-2033106270) in the module

You could cherry-pick the commit and then drop it during a rebase once the other PR gets merged.

technobaboo commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,118 @@
+    python3
+  ];
+
+  buildInputs = [
```
agreed, making this change made it work for me

technobaboo commented:
Everything works just fine for me with the lynx r1 and nixOS (besides video encoding settings issues) so yeah, ship it!

(it might be nice to put those video encoding settings in the config though if possible)

PassiveLemon commented:
> Everything works just fine for me with the lynx r1 and nixOS (besides video encoding settings issues) so yeah, ship it!
> 
> (it might be nice to put those video encoding settings in the config though if possible)

I don't see a reason to put those in the config. They will be used if they are enabled in WiVRn's config so all we need to do is just make sure they can be built

technobaboo commented:
is there anything left to do before this can be merged? it's proven to work and all

PassiveLemon commented:
Not that I know of. Just waiting for reviewers to do their thing

ckiee commented:
Approved with the assumption the recent pushes are just rebasing. I'll have a look at merging tomorrow (:

PassiveLemon commented:
> Approved with the assumption the recent pushes are just rebasing. I'll have a look at merging tomorrow (:

Yep, there was a conflict in the 24.05 release notes file so I fixed that.

awtterpip commented:
any timeline for this getting merged?

Scrumplex commented on nixos/doc/manual/release-notes/rl-2405.section.md:
```diff
@@ -171,7 +171,13 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 - [nh](https://github.com/viperML/nh), yet another Nix CLI helper. Available as [programs.nh](#opt-programs.nh.enable).
 
-- [ALVR](https://github.com/alvr-org/alvr), a VR desktop streamer. Available as [programs.alvr](#opt-programs.alvr.enable)
+- binfmt option for AppImage-run to support running [AppImage](https://appimage.org/)'s seamlessly on NixOS.. Available as [programs.appimage.binfmt](#opt-programs.appimage.binfmt).
+
+- [nh](https://github.com/viperML/nh), yet another Nix CLI helper. Available as [programs.nh](#opt-programs.nh.enable).
+
+- [ALVR](https://github.com/alvr-org/alvr), a VR desktop streamer. Available as [programs.alvr](#opt-programs.alvr.enable).
```
looks like you accidentally duplicated these changelog entries

Scrumplex commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+, avahi
+, boost
+, cmake
+, cudaPackages
```
```suggestion
, cudaPackages ? { }
```

Scrumplex commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+, vulkan-loader
+, vulkan-tools
+, x264
+}:
```
```suggestion
, cudaSupport ? config.cudaSupport
}:
```

Scrumplex commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+  buildInputs = [
+    avahi
+    boost
+    cudaPackages.cuda_cudart
```
```suggestion
```

Scrumplex commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+    vulkan-loader
+    vulkan-tools
+    x264
+  ];
```
```suggestion
  ] ++ lib.optionals cudaSupport [
    cudaPackages.cuda_cudart
  ];
```

Scrumplex reviewed (approved):
This would make it easier for people to use this package who don't have CUDA hardware

Pandapip1 commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+    (lib.cmakeBool "WIVRN_BUILD_CLIENT" false)
+    (lib.cmakeBool "WIVRN_USE_VAAPI" true)
+    (lib.cmakeBool "WIVRN_USE_X264" true)
+    (lib.cmakeBool "WIVRN_USE_NVENC" false)
```
Could you add a comment stating why NVENC is disabled?

Pandapip1 commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "WIVRN_BUILD_CLIENT" false)
```
I'd would like it if the client build was enabled and put in a seperate output, but I wouldn't consider this a blocker.

ckiee replied:
Indeed, not a blocker. The PC test client build seems to be an internal artifact and is absolutely not relevant for the server part, and is therefore out of scope for this PR.

Pandapip1 commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+    hash = "sha256-+RTHS9ShicuzhiAVAXf38V6k4SVr+Bc2xUjpRWZoB0c=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    cudaPackages.cuda_nvcc
+    git
+    pkg-config
+    python3
+  ];
```
I think @Scrumplex missed this reference to `cudaPackages`:

```suggestion
  nativeBuildInputs = [
    cmake
    git
    pkg-config
    python3
  ] ++ lib.optionals cudaSupport [
    cudaPackages.cuda_nvcc
  ];
```

Pandapip1 commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,112 @@
+    hash = "sha256-dnc9UNETDzT+sqo9bSTP1qZs/7kWftDo50yRgP94Mh4=";
+  };
+
+  monadoSrc = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "monado";
+    repo = "monado";
+    # Version stated in CMakeList for WiVRn 0.14.1
+    rev = "ffb71af26f8349952f5f820c268ee4774613e200";
+    hash = "sha256-+RTHS9ShicuzhiAVAXf38V6k4SVr+Bc2xUjpRWZoB0c=";
+  };
```
Any reason why you can't just depend o monado, or use the monado derivation's `src` attribute?

ckiee replied:
WiVRn vendors a monado source and builds it in its build system, we override `FETCHCONTENT_SOURCE_DIR_MONADO`.

Pandapip1 replied:
That makes sense. Thanks for clarifying!

Pandapip1 reviewed (changes requested):
I second @Scrumplex's request to make CUDA optional.

PassiveLemon commented:
I had forgotten to push changes I made a bit ago. Could you review the updated commits? I believe Cuda isn't required to build now

Goxore commented:
Tried building and running the wivrn-server binary, and it says that the version is 0.0.0.0, causing version mismatch error on the quest. Just me?

PassiveLemon commented:
Are you sure you installed the right version APK to your Quest? I've always seen v0.0.0 as well but only encountered version mismatch errors when the server and Quest were different versions.

Goxore commented:
I am using the 0.14.1 prebuilt release APK, which I assume to be compatible.
https://github.com/Meumeu/WiVRn/releases/tag/v0.14.1

Pandapip1 commented:
Result of `nixpkgs-review pr 299830` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wivrn</li>
  </ul>
</details>

PassiveLemon commented:
> I am using the 0.14.1 prebuilt release APK, which I assume to be compatible. https://github.com/Meumeu/WiVRn/releases/tag/v0.14.1

Not quite sure what your issue is then, I can't really help with that

ckiee commented:
make sure to uninstall first before `adb install`ing, it can be picky sometimes

fpletz committed:
glances: reformat and update dependencies

ckiee reviewed (approved):
Code LGTM, going for merge.
#### :white_check_mark: build

Result of `nixpkgs-review pr 299830` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wivrn</li>
  </ul>
</details>

#### checklist
- [X] server runs w/o nixos module (just avahi cfg)
- [X] server runs w nixos module
- [ ] needs testing w android client

PassiveLemon commented:
This push fixes Cuda not being found and makes cudaSupport optional for NVENC support

xytovl commented on pkgs/by-name/wi/wivrn/package.nix:
```diff
@@ -0,0 +1,151 @@
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "WIVRN_BUILD_CLIENT" false)
+    (lib.cmakeBool "WIVRN_USE_VAAPI" true)
+    (lib.cmakeBool "WIVRN_USE_X264" true)
+    (lib.cmakeBool "WIVRN_USE_NVENC" cudaSupport)
```
There are a few "automatic" options that should be set manually:
`WIVRN_USE_SYSTEM_BOOST` requires boost 1.75 for server, 1.84 for Wireshark plugin
`WIVRN_USE_SYSTEMD` for pretty hostname and to use a safer codepath when using monado
`WIVRN_USE_PIPEWIRE` / `WIVRN_USE_PULSEAUDIO` for audio backends

PassiveLemon commented:
Xytovl pointed out to me in Discord that the URLs for the patches are based on the master branch and that I should change it to the version tag. My problem is, I don't know how to reference the wivrn version attribute in the monadoSrc derivation. Using ${finalAttrs.version} will return the version specified for Monado. How should I go about this?

Pandapip1 commented:
> Xytovl pointed out to me in Discord that the URLs for the patches are based on the master branch and that I should change it to the version tag. My problem is, I don't know how to reference the wivrn version attribute in the monadoSrc derivation. Using ${finalAttrs.version} will return the version specified for Monado. How should I go about this?

Use `(monado.src + "/path/to/file")`.

Pandapip1 reviewed (approved):
Diff LGTM

Deric-W committed:
nixos/nextcloud-notify_push: use `Type=notify`

This prevents the post start script from running
before necessary sockets have been created.

It also prevents an unused shell from being kept around
by using `exec` to make `notify_push` the main process.

viraptor committed:
ejson2env: add tests

GaetanLepage committed:
python311Packages.dask: 2024.5.1 -> 2024.5.2

Diff: https://github.com/dask/dask/compare/refs/tags/2024.5.1...2024.5.2

Changelog: https://docs.dask.org/en/latest/changelog.html

GaetanLepage committed:
python311Packages.dask-expr: 1.1.1 -> 1.1.2

Diff: https://github.com/dask/dask-expr/compare/refs/tags/v1.1.1...v1.1.2

joshlupo committed:
session-desktop: file desktop exec path

wegank committed:
Merge pull request #313718 from r-ryantm/auto-update/clj-kondo

clj-kondo: 2024.03.13 -> 2024.05.24

wegank committed:
Merge pull request #314747 from r-ryantm/auto-update/obs-studio-plugins.obs-move-transition

obs-studio-plugins.obs-move-transition: 2.12.0 -> 3.0.0

wegank committed:
Merge pull request #314357 from r-ryantm/auto-update/aws-sam-cli

aws-sam-cli: 1.116.0 -> 1.117.0

SuperSandro2000 committed:
Merge pull request #315913 from Sigmanificient/wakatime-cli

wegank committed:
Merge pull request #313950 from r-ryantm/auto-update/vimPlugins.vim-clap

vimPlugins.vim-clap: 0.53 -> 0.54

wegank committed:
Merge pull request #314955 from r-ryantm/auto-update/linkerd_edge

linkerd_edge: 24.5.3 -> 24.5.5

Johny-LPM committed:
supergfxctl-plasmoid: init at 2.0.0 (#315196)

SuperSandro2000 committed:
Merge pull request #316558 from dotlambda/ccache

wegank committed:
Merge pull request #314930 from r-ryantm/auto-update/spire

spire: 1.9.5 -> 1.9.6

wegank committed:
Merge pull request #314967 from r-ryantm/auto-update/unciv

unciv: 4.11.13 -> 4.11.16

wegank committed:
Merge pull request #314928 from r-ryantm/auto-update/makima

makima: 0.8.4 -> 0.9.1

wegank committed:
Merge pull request #314905 from r-ryantm/auto-update/fable

fable: 4.13.0 -> 4.18.0

wegank committed:
Merge pull request #314903 from r-ryantm/auto-update/bacon

bacon: 2.18.0 -> 2.18.2

wegank committed:
Merge pull request #314878 from r-ryantm/auto-update/snappymail

snappymail: 2.36.1 -> 2.36.3

wegank committed:
Merge pull request #314879 from r-ryantm/auto-update/telescope

telescope: 0.9 -> 0.9.1

wegank committed:
Merge pull request #315086 from r-ryantm/auto-update/nu_scripts

nu_scripts: 0-unstable-2024-05-19 -> 0-unstable-2024-06-01

doronbehar committed:
Merge pull request #316609 from r-ryantm/auto-update/svg2pdf

svg2pdf: 0.10.0 -> 0.11.0

wegank committed:
Merge pull request #314985 from r-ryantm/auto-update/pgbackrest

pgbackrest: 2.51 -> 2.52

SuperSandro2000 committed:
Merge pull request #311358 from luftmensch-luftmensch/snipaste_2.9-Beta2

wegank committed:
Merge pull request #315083 from r-ryantm/auto-update/elasticmq-server-bin

elasticmq-server-bin: 1.6.1 -> 1.6.2

wegank committed:
Merge pull request #315155 from r-ryantm/auto-update/reaper

reaper: 7.15 -> 7.16

wegank committed:
Merge pull request #315232 from r-ryantm/auto-update/fission

fission: 1.20.1 -> 1.20.2

wegank committed:
Merge pull request #315185 from r-ryantm/auto-update/cemu

cemu: 2.0-82 -> 2.0-85

wegank committed:
Merge pull request #315227 from r-ryantm/auto-update/kube-router

kube-router: 2.1.2 -> 2.1.3

wegank committed:
Merge pull request #315226 from r-ryantm/auto-update/nodenv

nodenv: 1.4.1 -> 1.5.0

wegank committed:
Merge pull request #315222 from r-ryantm/auto-update/deck

deck: 1.37.0 -> 1.38.0

SuperSandro2000 committed:
Merge pull request #311934 from luftmensch-luftmensch/everdo_1.9.0

wegank committed:
Merge pull request #315228 from r-ryantm/auto-update/libnabo

libnabo: 1.1.1 -> 1.1.2

wegank committed:
Merge pull request #315188 from r-ryantm/auto-update/python311Packages.sounddevice

python311Packages.sounddevice: 0.4.6 -> 0.4.7

SuperSandro2000 committed:
Merge pull request #294108 from buckley310/2024-03-07-vmware

wegank committed:
Merge pull request #315056 from r-ryantm/auto-update/python311Packages.wagtail-factories

python311Packages.wagtail-factories: 4.1.0 -> 4.2.1

wegank committed:
Merge pull request #315662 from r-ryantm/auto-update/suitesparse-graphblas

suitesparse-graphblas: 9.1.0 -> 9.2.0

h7x4 committed:
Merge pull request #316648 from s1ls/invidious-router-descfix

nixos/invidious-router: remove redundant "Enables" in description

SuperSandro2000 committed:
Merge pull request #311903 from srhb/squid-exporter

SuperSandro2000 committed:
Merge pull request #311405 from luftmensch-luftmensch/dbgate_5.2.7

wegank committed:
Merge pull request #314962 from r-ryantm/auto-update/python311Packages.githubkit

python311Packages.githubkit: 0.11.4 -> 0.11.5

SuperSandro2000 committed:
Merge pull request #309447 from mevatron/mevatron/update-jcef-to-241

gador committed:
pgadmin4: 8.6 -> 8.7

Signed-off-by: Florian Brandes <florian.brandes@posteo.de>

wegank committed:
Merge pull request #314904 from r-ryantm/auto-update/python311Packages.google-cloud-bigquery

python311Packages.google-cloud-bigquery: 3.23.0 -> 3.23.1

wegank committed:
Merge pull request #314783 from r-ryantm/auto-update/jellyfin-mpv-shim

jellyfin-mpv-shim: 2.6.0 -> 2.7.0.post2

wegank committed:
Merge pull request #314958 from r-ryantm/auto-update/python311Packages.rapidgzip

python311Packages.rapidgzip: 0.13.3 -> 0.14.2

wegank committed:
Merge pull request #313880 from r-ryantm/auto-update/spdx-license-list-data

spdx-license-list-data: 3.23 -> 3.24.0

wegank committed:
Merge pull request #313679 from r-ryantm/auto-update/keepalived

keepalived: 2.2.8 -> 2.3.0

7c6f434c committed:
Merge pull request #315896 from hraban/sbcl/2.4.5

sbcl: 2.4.4 -> 2.4.5

wegank committed:
Merge pull request #313346 from r-ryantm/auto-update/ipxe

ipxe: 1.21.1-unstable-2024-04-17 -> 1.21.1-unstable-2024-05-31

wegank committed:
Merge pull request #313282 from r-ryantm/auto-update/ecs-agent

ecs-agent: 1.82.3 -> 1.82.4

drupol committed:
Merge pull request #316394 from Aleksanaa/nixd

nixd: 2.1.2 -> 2.2.0; {nixf,nixt}: init

drupol committed:
Merge pull request #316022 from heisfer/eta

eta: init at 1.0.1

wegank committed:
Merge pull request #313479 from r-ryantm/auto-update/qxmpp

qxmpp: 1.6.1 -> 1.7.0

wegank committed:
Merge pull request #314177 from r-ryantm/auto-update/python311Packages.sphinxcontrib-plantuml

python311Packages.sphinxcontrib-plantuml: 0.29 -> 0.30

wegank committed:
Merge pull request #314118 from r-ryantm/auto-update/python311Packages.tempest

python311Packages.tempest: 38.0.0 -> 39.0.0

wegank committed:
Merge pull request #313065 from r-ryantm/auto-update/python311Packages.bdffont

python311Packages.bdffont: 0.0.24 -> 0.0.25

wegank committed:
Merge pull request #314032 from r-ryantm/auto-update/python311Packages.pytorch-lightning

python311Packages.pytorch-lightning: 2.2.4 -> 2.2.5

wegank committed:
Merge pull request #314187 from r-ryantm/auto-update/python311Packages.wallbox

python311Packages.wallbox: 0.6.0 -> 0.7.0

wegank committed:
Merge pull request #314412 from r-ryantm/auto-update/python311Packages.azure-mgmt-netapp

python311Packages.azure-mgmt-netapp: 12.0.0 -> 13.0.0

drupol committed:
Merge pull request #315642 from savyajha/qbittorrent-api

python311Packages.qbittorrent-api: 2024.3.60 -> 2024.5.63

drupol committed:
Merge pull request #313145 from DontEatOreo/pkgs-win-disk-writer

win-disk-writer: init at 1.3

wegank committed:
Merge pull request #314770 from r-ryantm/auto-update/python311Packages.robotframework-seleniumlibrary

python311Packages.robotframework-seleniumlibrary: 6.3.0 -> 6.4.0

drupol committed:
Merge pull request #311646 from GaetanLepage/pytensor

python311Packages.pytensor: 2.20.0 -> 2.22.1, python311Packages.pymc: 5.15.0 -> 5.15.1

wegank committed:
Merge pull request #313040 from r-ryantm/auto-update/ventoy

ventoy: 1.0.97 -> 1.0.98

drupol committed:
Merge pull request #316535 from avnik/avnik/wine-update

wine64Packages.{unstable,staging}: 9.9 -> 9.10

thillux committed:
jitterentropy: 3.4.1 -> 3.5.0

Changes:
- add distinction between intermittent and permanent health failure
- add compile time option to allow configuring a mask to reduce the size of the time stamp used for the APT

Upstream changelog: https://github.com/smuellerDD/jitterentropy-library/releases/tag/v3.5.0

Signed-off-by: Markus Theil <theil.markus@gmail.com>

r-ryantm committed:
minio: 2024-05-27T19-17-46Z -> 2024-05-28T17-19-04Z

eclairevoyant committed:
Merge pull request #316704 from babariviere/chore/remove-babariviere

remove 'babariviere' as a maintainer

azahi committed:
Merge pull request #316677 from r-ryantm/auto-update/vcmi

vcmi: 1.5.1 -> 1.5.2

jonasnick committed:
lnd: 0.17.5-beta -> 0.18.0-beta

rhoriguchi committed:
nixos/adguardhome: fix typo

dotlambda committed:
Merge pull request #316238 from r-ryantm/auto-update/python311Packages.ytmusicapi

python311Packages.ytmusicapi: 1.7.2 -> 1.7.3

SuperSandro2000 committed:
Merge pull request #316125 from Nanotwerp/kitty-up

drupol committed:
Merge pull request #316686 from r-ryantm/auto-update/renderdoc

renderdoc: 1.32 -> 1.33

drupol committed:
Merge pull request #316556 from r-ryantm/auto-update/kent

kent: 446 -> 465

drupol committed:
Merge pull request #316575 from r-ryantm/auto-update/keepass

keepass: 2.56 -> 2.57

drupol committed:
Merge pull request #316576 from r-ryantm/auto-update/oil

oil: 0.21.0 -> 0.22.0

drupol committed:
Merge pull request #316550 from r-ryantm/auto-update/logrotate

logrotate: 3.21.0 -> 3.22.0

SuperSandro2000 committed:
Merge pull request #314683 from trofi/diffoscope-update

physics-enthusiast committed:
python3Packages.libnbd: init at 1.18.2 (#313358)

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

fabaff committed:
Merge pull request #316613 from fabaff/metasploit-bump

metasploit: 6.4.10 -> 6.4.11

drupol committed:
Merge pull request #316421 from markuskowa/upd-rdma-core

rdma-core: 51.0 -> 52.0

drupol committed:
Merge pull request #316420 from r-ryantm/auto-update/indilib

indilib: 2.0.7 -> 2.0.8

RoyDubnium committed:
ollama: 0.1.39 -> 0.1.41

jian-lin committed:
Merge pull request #316625 from brahyerr/el-easydraw

emacsPackage.el-easydraw: 1.1.0 → 20240529.1009

SuperSandro2000 committed:
Merge pull request #316722 from rhoriguchi/nixos/adguardhome

SuperSandro2000 committed:
Merge pull request #293032 from afh/update-refactor-dockutil

SuperSandro2000 committed:
nixos/tailscale-auth: fix enable option description

SuperSandro2000 committed:
Merge pull request #316636 from palasso/patch-1

SuperSandro2000 committed:
Merge pull request #314234 from me-and/git-update-script

prusnak committed:
Merge pull request #316721 from jonasnick/lnd-0.18.0

SuperSandro2000 committed:
Merge pull request #316023 from tie/puppet-bolt-gemconfig

SuperSandro2000 committed:
Merge pull request #308725 from duament/fcitx5-dbus-substitute

drupol committed:
Merge pull request #316710 from gador/pgadmin-8.7

pgadmin4: 8.6 -> 8.7

SuperSandro2000 committed:
Merge pull request #299995 from tie/centrifugo-tests-fix-shards

SuperSandro2000 committed:
Merge pull request #316358 from majewsky/portunus-remove-libxcrypt-legacy

drupol committed:
Merge pull request #316671 from r-ryantm/auto-update/podman-tui

podman-tui: 1.0.1 -> 1.1.0

drupol committed:
Merge pull request #316612 from r-ryantm/auto-update/armadillo

armadillo: 12.8.3 -> 12.8.4

drupol committed:
Merge pull request #316605 from r-ryantm/auto-update/trezor-suite

trezor-suite: 24.5.2 -> 24.5.3

drupol committed:
Merge pull request #316602 from r-ryantm/auto-update/uwsgi

uwsgi: 2.0.25.1 -> 2.0.26

drupol committed:
Merge pull request #316603 from r-ryantm/auto-update/libmediainfo

libmediainfo: 24.04 -> 24.05

drupol committed:
Merge pull request #316594 from wrvsrx/fix-cuda-alias-warning

cudaPackages: fix cuda alias warning

drupol committed:
Merge pull request #316555 from r-ryantm/auto-update/scid-vs-pc

scid-vs-pc: 4.24 -> 4.25

drupol committed:
Merge pull request #316554 from r-ryantm/auto-update/prooftree

prooftree: 0.13 -> 0.14

drupol committed:
Merge pull request #316559 from r-ryantm/auto-update/ocamlPackages.cmdliner

ocamlPackages.cmdliner: 1.2.0 -> 1.3.0

drupol committed:
Merge pull request #316560 from r-ryantm/auto-update/bootstrap-studio

bootstrap-studio: 6.6.1 -> 6.7.0

drupol committed:
Merge pull request #316546 from r-ryantm/auto-update/sitespeed-io

sitespeed-io: 33.6.0 -> 34.0.1

SuperSandro2000 committed:
Merge pull request #316581 from dotlambda/nextcloud

SuperSandro2000 committed:
Merge pull request #315533 from Deric-W/notify_push

SomeoneSerge committed:
Merge pull request #316516 from aidalgol/libxnvctrl-nixos-fix

nixos/nvidia: add missing check for nvidia settings enabled

useranon350 committed:
xfce.xfce4-sensors-plugin: Enable xnvctrl by default

Have to be explicitly enabled since this tries to figure out the default
based on the existence of a hardcoded `/usr/include/NVCtrl` path, though
it looks like `NIX_CFLAGS_COMPILE` is properly (?) set so we don't need to
further patch it.

Format with nixfmt-rfc-style at the same time.

SuperSandro2000 committed:
Merge pull request #307690 from arikgrahl/setup-envtest

SuperSandro2000 committed:
Merge pull request #242114 from jn-sena/everforest-gtk-init

SuperSandro2000 committed:
Merge pull request #264635 from baloo/baloo/amazon-image/pkg-reimport

SuperSandro2000 committed:
Merge pull request #293313 from afh/update-apparency

SuperSandro2000 committed:
Merge pull request #311688 from onemoresuza/hare-fix-mime-module

SuperSandro2000 committed:
Merge pull request #311324 from pedohorse/houdini_20-0-506_20-0-688

SuperSandro2000 committed:
Merge pull request #308119 from jopejoe1/boost-process

SuperSandro2000 committed:
Merge pull request #300562 from maxbrunet/feat/tanka/fish-zsh-shell-completions

SuperSandro2000 committed:
Merge pull request #303338 from jchw-forks/86box-autoupdate

SuperSandro2000 committed:
Merge pull request #306787 from Guanran928/fcitx5-themes

mweinelt committed:
zigbee2mqtt: 1.37.1 -> 1.38.0

https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.38.0

SuperSandro2000 committed:
Merge pull request #307287 from illustris/proxmox-vma

illustris committed:
nixos/proxmox-lxc: fix console access (#307163)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

mweinelt committed:
Merge pull request #315887 from mweinelt/mushroom-3.6.0

home-assistant-custom-lovelace-modules.mushroom: 3.5.5 -> 3.6.0

SuperSandro2000 committed:
Merge pull request #315532 from GaetanLepage/certomancer

r-ryantm committed:
volatility3: 2.5.2 -> 2.7.0

SomeoneSerge committed:
Merge pull request #306716 from ShamrockLee/apptainer-default-path-substitution

apptainer, singularity: refactor defaultPath substitution

SomeoneSerge committed:
Merge pull request #312985 from GaetanLepage/keras

python311Packages.keras: 3.2.1 -> 3.3.3

eclairevoyant committed:
Merge pull request #312119 from hogcycle/yoink

yoink: init at 0.5.0

fpletz committed:
Merge pull request #312521 from jstsmthrgk/update-emptyepsilon

r-ryantm committed:
ast-grep: 0.22.4 -> 0.22.5

r-ryantm committed:
icloudpd: 1.18.0 -> 1.19.0

fpletz committed:
Merge pull request #316211 from r-ryantm/auto-update/glances

r-ryantm committed:
flawz: 0.2.0 -> 0.2.1

r-ryantm committed:
gomplate: 3.11.7 -> 3.11.8

r-ryantm committed:
osl: 1.13.9.0 -> 1.13.10.0

r-ryantm committed:
gitui: 0.26.2 -> 0.26.3

viraptor committed:
Merge pull request #316029 from viraptor/ejson2env-tests

ejson2env: add tests

r-ryantm committed:
ticker: 4.6.2 -> 4.6.3

r-ryantm committed:
steamguard-cli: 0.13.0 -> 0.14.0

r-ryantm committed:
nh: 3.5.15 -> 3.5.16

JerrySM64 committed:
linuxKernel.kernels.linux_zen: 6.9.2-zen1 -> 6.9.3-zen1

JerrySM64 committed:
linuxKernel.kernels.linux_lqx: 6.8.11-lqx1 -> 6.9.3-lqx1

flokli committed:
Merge pull request #316697 from bbenno/fix-gollum

nixos/gollum: fix systemd tempfile permission

drupol committed:
Merge pull request #316803 from r-ryantm/auto-update/nh

nh: 3.5.15 -> 3.5.16

drupol committed:
Merge pull request #316498 from joshlupo/patch-1

session-desktop: file desktop exec path

drupol committed:
trezor-suite: update hash for `aarch64-linux`

Context: https://github.com/NixOS/nixpkgs/pull/316605

bachp committed:
Merge pull request #316728 from r-ryantm/auto-update/minio

minio: 2024-05-27T19-17-46Z -> 2024-05-28T17-19-04Z

fabaff committed:
python311Packages.reolink-aio: 0.8.11 -> 0.9.0

Diff: https://github.com/starkillerOG/reolink_aio/compare/refs/tags/0.8.11...0.9.0

Changelog: https://github.com/starkillerOG/reolink_aio/releases/tag/0.9.0

fabaff committed:
python312Packages.tencentcloud-sdk-python: 3.0.1159 -> 3.0.1160

Diff: https://github.com/TencentCloud/tencentcloud-sdk-python/compare/refs/tags/3.0.1159...3.0.1160

Changelog: https://github.com/TencentCloud/tencentcloud-sdk-python/blob/3.0.1160/CHANGELOG.md

fabaff committed:
usort: 1.0.7 -> 1.0.8

Diff: https://github.com/facebook/usort/compare/refs/tags/v1.0.7...v1.0.8

Changelog: https://github.com/facebook/usort/blob/1.0.8/CHANGELOG.md

datafoo committed:
vscode-extensions.streetsidesoftware.code-spell-checker: 4.0.1 -> 4.0.2

drupol committed:
Merge pull request #316729 from RoyDubnium/master

ollama: 0.1.39 -> 0.1.41

jnsgruk committed:
Merge pull request #316779 from r-ryantm/auto-update/icloudpd

K900 committed:
Merge pull request #316655 from raboof/grub2-deterministic-locales

grub2: make localization resources deterministic

numinit committed:
pkcs11-provider: add nix-update-script

drupol committed:
Merge pull request #316459 from Nydragon/rofi-obsidian-bump-version

rofi-obsidian: 0.1.0 -> 0.1.5

srhb committed:
nixos/garage: fix replication 1.0 assertion

Use the `cfg.package.version` (string) instead of the entire package so
users don't see  `error: value is a set while a string was expected`
instead of the intended assertion message.

prusnak committed:
Merge pull request #316808 from drupol/trezor-suite/erratum/pr-316605

trezor-suite: update hash for `aarch64-linux`

drupol committed:
Merge pull request #316819 from datafoo/vscode-extensions.streetsidesoftware.code-spell-checker

vscode-extensions.streetsidesoftware.code-spell-checker: 4.0.1 -> 4.0.2

flokli committed:
Merge pull request #316836 from srhb/fix-garage-assert

nixos/garage: fix replication 1.0 assertion

7c6f434c committed:
Merge pull request #316221 from teto/matt/lua-add-luarocks

luaPackages.luarocks: generate luarocks as well

kirillrdy committed:
Merge pull request #315882 from MikaelFangel/update-fantomas

fantomas: 6.3.4 -> 6.3.7

r-ryantm committed:
python311Packages.model-bakery: 1.18.0 -> 1.18.1

r-ryantm committed:
python311Packages.etils: 1.8.0 -> 1.9.0

r-ryantm committed:
python311Packages.meep: 1.28.0 -> 1.29.0

pbsds committed:
Merge pull request #316702 from GaetanLepage/tuigreet

greetd.tuigreet: 0.9.0 -> 0.9.1

matthiasbeyer committed:
Merge pull request #316379 from r-ryantm/auto-update/cargo-dist

cargo-dist: 0.14.1 -> 0.15.0

matthiasbeyer committed:
cargo-llvm-cov: 0.6.9 -> 0.6.10

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

r-ryantm committed:
github-runner: 2.316.1 -> 2.317.0 (#316806)

matthiasbeyer committed:
Merge pull request #316858 from matthiasbeyer/update-cargo-llvm-cov

cargo-llvm-cov: 0.6.9 -> 0.6.10

Commit:
uiua: 0.10.3 -> 0.11.0

Commit:
vscode-extensions.uiua-lang.uiua-vscode: 0.0.42 -> 0.0.44

r-ryantm committed:
tftui: 0.13.2 -> 0.13.3

r-ryantm committed:
python311Packages.pyomo: 6.7.2 -> 6.7.3

markuskowa committed:
Merge pull request #316846 from r-ryantm/auto-update/python311Packages.meep

python311Packages.meep: 1.28.0 -> 1.29.0

SuperSandro2000 committed:
Merge pull request #316732 from SuperSandro2000/tailscale-auth

mweinelt committed:
Merge pull request #316747 from mweinelt/zigbee2mqtt-1.38.0

zigbee2mqtt: 1.37.1 -> 1.38.0

ncfavier committed:
mueval: remove check

Removes the check introduced in https://github.com/NixOS/nixpkgs/pull/301484
as it fails too often on resource-constrained machines.

malt3 committed:
bazel_7: 7.1.0 -> 7.1.2 (#314279)

* bazel_7: 7.1.0 -> 7.1.2

* bazel_7: update lockfiles

* bazel_7: upgrade lockfile format for versions > 3

The new lockfile format 6 drops the "name" attribute of each repoSpec.
See also: https://github.com/bazelbuild/bazel/pull/21026
This prevents the builder from effectively the deps using
requiredDepNamePredicate.
Instead, we now generate names from other metadata.

r-ryantm committed:
retroarch-assets: 1.17.0-unstable-2024-04-24 -> 1.19.0-unstable-2024-05-30

r-ryantm committed:
libretro.play: unstable-2024-05-27 -> unstable-2024-05-28

r-ryantm committed:
libretro.pcsx-rearmed: unstable-2024-05-20 -> unstable-2024-05-30

asymmetric committed:
Merge pull request #316465 from gmacon/tarsnapper-pytest-8

tarsnapper: use nose to run tests

ncfavier committed:
Merge pull request #316886 from ncfavier/mueval-no-check

mueval: remove check

r-ryantm committed:
libretro.mame2003: unstable-2024-05-27 -> unstable-2024-06-01

r-ryantm committed:
libretro.ppsspp: unstable-2024-05-28 -> unstable-2024-06-03

r-ryantm committed:
libretro.mame2003-plus: unstable-2024-05-28 -> unstable-2024-05-29

superherointj committed:
Merge pull request #316865 from Defelo/uiua

uiua: 0.10.3 -> 0.11.0

tomberek committed:
Merge pull request #315516 from numinit/pkcs11-provider-0.4

pkcs11-provider: 0.3 -> 0.4

pbsds committed:
Merge pull request #316822 from GaetanLepage/tf-keras

python311Packages.tf-keras: init at 2.16.0

pbsds committed:
Merge pull request #316222 from pbsds/bump-gradio-pdf-1717177749

python311Packages.gradio: 4.29.0 -> 4.32.1

drupol committed:
Merge pull request #315844 from aciceri/zmkbatx

simpleBluez: init at 0.7.3, simpleDBus: init at 0.7.3, zmkBATx: init at 1.0.1

rnhmjoj committed:
Merge pull request #312317 from tom-kuca/replace-existing-network-addresses

nixos/network-interfaces: prevent failure when a network address already exists

Lassulus committed:
Merge pull request #300915 from MayNiklas/patch-2

papermc: fix update script

NickCao committed:
Merge pull request #316527 from r-ryantm/auto-update/qownnotes

qownnotes: 24.5.8 -> 24.6.0

NickCao committed:
Merge pull request #316698 from r-ryantm/auto-update/xpipe

xpipe: 9.3 -> 9.4

NickCao committed:
Merge pull request #316628 from DanielSidhion/clips-update

clips: 6.40 -> 6.4.1

NickCao committed:
Merge pull request #316348 from r-ryantm/auto-update/dart-sass

dart-sass: 1.77.2 -> 1.77.4

NickCao committed:
Merge pull request #316328 from r-ryantm/auto-update/eigenlayer

eigenlayer: 0.8.0 -> 0.8.1

NickCao committed:
Merge pull request #316317 from r-ryantm/auto-update/rain

rain: 1.9.0 -> 1.10.0

r-ryantm committed:
libretro.fbneo: unstable-2024-05-28 -> unstable-2024-06-02

SuperSandro2000 committed:
nixos/installation-cd-plasma5: don't use alias for konsole

K900 committed:
Merge pull request #316929 from SuperSandro2000/cd-konsole

nixos/installation-cd-plasma5: don't use alias for konsole

SuperSandro2000 committed:
nixos/installation-cd-plasma5: fix even more aliases

Aman9das committed:
bitwarden-menu: refactor

K900 committed:
Merge pull request #316933 from SuperSandro2000/cd-plasma-aliases

nixos/installation-cd-plasma5: fix even more aliases

natsukium committed:
Merge pull request #316256 from GaetanLepage/dask

python311Packages.dask: 2024.5.1 -> 2024.5.2

fabaff committed:
volatility3: refactor

natsukium committed:
Merge pull request #316873 from r-ryantm/auto-update/python311Packages.pyomo

python311Packages.pyomo: 6.7.2 -> 6.7.3

natsukium committed:
Merge pull request #316856 from r-ryantm/auto-update/python311Packages.etils

python311Packages.etils: 1.8.0 -> 1.9.0

natsukium committed:
Merge pull request #316843 from r-ryantm/auto-update/python311Packages.model-bakery

python311Packages.model-bakery: 1.18.0 -> 1.18.1

nikstur committed:
Merge pull request #316716 from thillux/jitterentropy-3.5.0

jitterentropy: 3.4.1 -> 3.5.0

NickCao committed:
Merge pull request #316154 from r-ryantm/auto-update/freeipa

freeipa: 4.11.1 -> 4.12.0

NickCao committed:
Merge pull request #316140 from r-ryantm/auto-update/gnustep.make

gnustep.make: 2.9.1 -> 2.9.2

NickCao committed:
Merge pull request #316055 from r-ryantm/auto-update/bkcrack

bkcrack: 1.6.1 -> 1.7.0

bachp committed:
Merge pull request #316777 from r-ryantm/auto-update/ast-grep

ast-grep: 0.22.4 -> 0.22.5

bachp committed:
Merge pull request #316755 from r-ryantm/auto-update/volatility3

volatility3: 2.5.2 -> 2.7.0

NickCao committed:
Merge pull request #315776 from r-ryantm/auto-update/reaction

reaction: 1.3.0 -> 1.3.1

NickCao committed:
Merge pull request #314129 from newAM/probe-rs-0.24.0

probe-rs: 0.23.0 -> 0.24.0; rename

NickCao committed:
Merge pull request #316798 from r-ryantm/auto-update/steamguard-cli

steamguard-cli: 0.13.0 -> 0.14.0

NickCao committed:
Merge pull request #316797 from r-ryantm/auto-update/ticker

ticker: 4.6.2 -> 4.6.3

NickCao committed:
Merge pull request #316793 from r-ryantm/auto-update/osl

osl: 1.13.9.0 -> 1.13.10.0

NickCao committed:
Merge pull request #316791 from r-ryantm/auto-update/gitui

gitui: 0.26.2 -> 0.26.3

NickCao committed:
Merge pull request #316790 from r-ryantm/auto-update/gomplate

gomplate: 3.11.7 -> 3.11.8

NickCao committed:
Merge pull request #316789 from r-ryantm/auto-update/flawz

flawz: 0.2.0 -> 0.2.1

NickCao committed:
Merge pull request #316817 from fabaff/usort-bump

usort: 1.0.7 -> 1.0.8

NickCao committed:
Merge pull request #316816 from fabaff/tencentcloud-sdk-python-bump

python312Packages.tencentcloud-sdk-python: 3.0.1159 -> 3.0.1160

NickCao committed:
Merge pull request #316813 from fabaff/reolink-aio-bump

python311Packages.reolink-aio: 0.8.11 -> 0.9.0

NickCao committed:
Merge pull request #316871 from r-ryantm/auto-update/tftui

tftui: 0.13.2 -> 0.13.3

bobby285271 committed:
Merge pull request #316744 from useranon350/xfce-plugin-temp

xfce.xfce4-sensors-plugin: Enable xnvctrl by default

SuperSandro2000 committed:
Merge pull request #316932 from Aman9das/patch-2

natsukium committed:
Merge pull request #316629 from natsukium/diffusers/update

python311Packages.diffusers: 0.27.2 -> 0.28.0

thiagokokada committed:
Merge pull request #316890 from r-ryantm/auto-update/retroarch-assets

retroarch-assets: 1.17.0-unstable-2024-04-24 -> 1.19.0-unstable-2024-05-30

thiagokokada committed:
Merge pull request #316902 from r-ryantm/auto-update/libretro.mame2003

libretro.mame2003: unstable-2024-05-27 -> unstable-2024-06-01

thiagokokada committed:
Merge pull request #316904 from r-ryantm/auto-update/libretro.ppsspp

libretro.ppsspp: unstable-2024-05-28 -> unstable-2024-06-03

thiagokokada committed:
Merge pull request #316905 from r-ryantm/auto-update/libretro.mame2003-plus

libretro.mame2003-plus: unstable-2024-05-28 -> unstable-2024-05-29

thiagokokada committed:
Merge pull request #316926 from r-ryantm/auto-update/libretro.fbneo

libretro.fbneo: unstable-2024-05-28 -> unstable-2024-06-02

thiagokokada committed:
Merge pull request #316894 from r-ryantm/auto-update/libretro.pcsx-rearmed

libretro.pcsx-rearmed: unstable-2024-05-20 -> unstable-2024-05-30

thiagokokada committed:
Merge pull request #316892 from r-ryantm/auto-update/libretro.play

libretro.play: unstable-2024-05-27 -> unstable-2024-05-28

roberth committed:
Merge pull request #312442 from fgaz/changelog-d/1.0

changelog-d: 0.1-git-2816ddb -> 1.0

thiagokokada committed:
Merge pull request #316804 from JerrySM64/update-kernels

linuxKernel.kernels.linux_zen: 6.9.2-zen1 -> 6.9.3-zen1 ; linuxKernel.kernels.linux_lqx: 6.8.11-lqx1 -> 6.9.3-lqx1

PassiveLemon committed:
nixos/wivrn: init module

PassiveLemon committed:
Merge branch 'wivrn' of https://github.com/PassiveLemon/nixpkgs into wivrn

PassiveLemon commented:
uh what

PassiveLemon commented:
I swear I just did a simple rebase

drupol commented:
It looks like you accidentally mass-pinged a bunch of people, which are now subscribed
and getting notifications for everything in this pull request. Unfortunately, they
cannot be automatically unsubscribed from the issue (removing review request does not
unsubscribe), therefore development cannot continue in this pull request anymore.

Please create a new pull request, link back to this one and ping the
people actually involved in here over there. For the next time, remember to set your PR to draft status before rebasing. In draft status, you can preview the list of maintainers that are about to be requested for review, which allows you to sidestep this issue.

In order to avoid this in the future, there are instructions for how to properly
rebase between branches in our [contribution guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging).
Setting your pull request to draft prior to rebasing is strongly recommended.
In draft status, you can preview the list of people that are about to be requested
for review, which allows you to sidestep this issue.
This is not a bulletproof method, though, as OfBorg still does review requests even on draft PRs.
